reference/test/reorder/mc64_kernels.cpp

```diff
@@ -156,6 +161,20 @@ class Mc64 : public ::testing::Test {
                               gko::as<perm_type>(result->get_operators()[1]));
     }
 
+    void assert_array_near(const gko::array<real_type>& a,
+                           const gko::array<real_type>& b, std::string name)
+    {
+        ASSERT_EQ(a.get_num_elems(), b.get_num_elems());
+        for (gko::size_type i = 0; i < a.get_num_elems(); i++) {
+            if (std::isfinite(a.get_const_data()[i]) ||
+                std::isfinite(b.get_const_data()[i])) {
+                ASSERT_NEAR(a.get_const_data()[i], b.get_const_data()[i],
+                            r<value_type>::value)
+                    << name << '[' << i << ']';
+            }
```

Comment:
```suggestion
            } else {
                ASSERT_FALSE(true) << "not comparable " <<  name << '[' << i << ']';
            }
```
it's not the case for the tests though.

Reply:
inf == inf is fine

Reply:
NaN vs Inf should be wrong or NaN vs Nan